src/env_universal_common.rs

```diff
@@ -1,41 +1,28 @@
 #![allow(clippy::bad_bit_mask)]
 
 use crate::common::{
-    str2wcstring, timef, unescape_string, valid_var_name, wcs2zstring, UnescapeFlags,
-    UnescapeStringStyle,
+    unescape_string, valid_var_name, wcs2zstring, UnescapeFlags, UnescapeStringStyle,
 };
 use crate::env::{EnvVar, EnvVarFlags, VarTable};
-use crate::fallback::fish_mkstemp_cloexec;
-use crate::fds::{open_cloexec, wopen_cloexec};
 use crate::flog::{FLOG, FLOGF};
+use crate::fs::{lock_and_load, rewrite_via_temporary_file};
+use crate::global_safety::RelaxedAtomicBool;
 use crate::path::path_get_config;
 use crate::path::{path_get_config_remoteness, DirRemoteness};
 use crate::wchar::{decode_byte_from_char, prelude::*};
-use crate::wcstringutil::{join_strings, string_suffixes_string, LineIterator};
-use crate::wutil::{
-    file_id_for_file, file_id_for_path, file_id_for_path_narrow, wdirname, wrealpath, wrename,
-    wstat, wunlink, FileId, INVALID_FILE_ID,
-};
-use errno::{errno, Errno};
-use libc::{EINTR, LOCK_EX};
-use nix::{fcntl::OFlag, sys::stat::Mode};
+use crate::wcstringutil::{join_strings, LineIterator};
+use crate::wutil::{file_id_for_file, file_id_for_path_narrow, wrealpath, FileId, INVALID_FILE_ID};
 use std::collections::hash_map::Entry;
 use std::collections::HashSet;
 use std::ffi::CString;
 use std::fs::File;
 use std::io::{Read, Write};
 use std::mem::MaybeUninit;
-use std::os::fd::AsRawFd;
-use std::os::unix::prelude::MetadataExt;
-
-// Pull in the O_EXLOCK constant if it is defined, otherwise set it to 0.
-#[cfg(any(apple, bsd))]
-const O_EXLOCK: OFlag = OFlag::O_EXLOCK;
 
-#[cfg(not(any(apple, bsd)))]
-const O_EXLOCK: OFlag = OFlag::empty();
+static LOCK_UVAR_FILE: RelaxedAtomicBool = RelaxedAtomicBool::new(true);
```

Comment:
I think this can remain a bool inside `EnvUniversal`, since we only use it from EnvUniversal's methods.
(We have lots of globals that we should slowly move away from.)
I guess that would make the call to `lock_file_path()` more awkward, because we can't use a mutable reference directly.
How about

```rust
let mut locking_enabled = self.locking_enable;
match lock_and_load(self.vars_path, &mut locking_enabled) { ... }
self.locking_enable = locking_enabled;
```

though if we're doing that, I guess we could also pass it by value and return value.. not sure

we can also clean this up later